packages/mermaid/src/diagram-api/diagram-orchestration.spec.ts

```diff
@@ -0,0 +1,45 @@
+import { it, describe, expect } from 'vitest';
+import { detectType } from './detectType';
+import { addDiagrams } from './diagram-orchestration';
+
+describe('diagram-orchestration', () => {
+  it('should register diagrams', () => {
+    expect(() => detectType('graph TD; A-->B')).toThrow();
+    addDiagrams();
+    expect(detectType('graph TD; A-->B')).toBe('flowchart');
+  });
+
+  describe('proper diagram types should be detetced', () => {
+    beforeAll(() => {
+      addDiagrams();
+    });
+
+    it.each([
+      { text: 'graph TD;', expected: 'flowchart' },
+      { text: 'flowchart TD;', expected: 'flowchart-v2' },
```

Comment:
Should this be `flowchart-v2` or `flowchart`?

Sorry, I'm not really an expert on many of the different mermaid diagram types!

Reply:
I was also confused, I've updated the tests after consulting @knsv.